gdal/frmts/jpegxr/jpegxrdataset.cpp

```diff
@@ -215,12 +215,23 @@ struct JPEGXRDecoder final
         if (nErr != WMP_errSuccess || !pImageDecode)
             throw JPEGXRException(nErr);
 
+        PKPixelInfo pi;
+        GetPixelInfo(pi);
+
+        // Alpha
+        // 0: Decode without alpha channel
+        // 1: Decode only alpha channel (TODO: unused, might be controlled with driver option)
+        // 2: Decode image & alpha (default)
+        if (!!(pi.grBit & PK_pixfmtHasAlpha))
```

Comment:
@rouault Heh, you read my mind. I actually contemplated for a moment that I should change this condition :) Will do.